src/iam/iam-base.ts

```diff
@@ -100,11 +105,11 @@ export class IAMBase {
   protected _safeAddress: string | undefined;
   protected _didSigner: IdentityOwner | undefined;
   protected _transactionOverrides: TransactionOverrides = {};
-  protected _providerType: WalletProvider | undefined;
-  protected _publicKey: string | undefined;
+  protected _providerType: WalletProvider;
+  protected _publicKey: string;
```

Comment:
I think both of them should by `type | null` cause when something is not stored in sessionStorage the browser API will return null

Reply:
Ok